caddyhttp/httpserver/server.go

```diff
@@ -333,6 +334,12 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Server", "Caddy")
 
+	if requestid.IsActive {
+		reqid := requestid.UUID()
+		c := context.WithValue(r.Context(), caddy.CtxKey("REQUESTID"), reqid)
+		r = r.WithContext(c)
+	}
```

Comment:
... actually, I just realized that setting a value on the replacer isn't probably going to work since, once the directive is moved up to the beginning of the chain, it's quite probable that the ResponseWriter won't be a Replacer type (yet)... until, for example, the `log` middleware is executed.

So perhaps you should ignore that advice for now, continue on with using context.